### ISSUE TITLE
Invalid isset operations variables existence violation detection.

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#870]
This code triggers the "Isset operations variables existence" inspection on the `isset($nextPrevious)` code fragment. This is incorrect because the code is valid.

```php
do {
    $previous = isset($nextPrevious) ? $nextPrevious : $this->getPreviousCollection();
} while ($previous && $nextPrevious = $previous->getPreviousCollection());
```

**Comment by kalessil** [comment]
Huh, that's a tough case. An absolutely valid case, but I'd like people to look into implicit declarations.
Hence, the only what I'll do is add documentation, promoting following:

```php
$nextPrevious = null;
do {
    $previous = $nextPrevious === null ? $this->getPreviousCollection() : $nextPrevious;
} while ($previous && $nextPrevious = $previous->getPreviousCollection());
```

PS: sorry, hope you understand my dilemma.

**Comment by Bilge** [comment]
No, sorry, I don't understand and I don't expect your inspections demand I rewrite my code to suit your personal preferences.

**Comment by kalessil** [comment]
Fair enough. Which plugin/version and IDE/version is it - I'll check what I can do.

**Comment by Bilge** [comment]
PhpStorm 2017.3, PHPEA 2.3.18

**Comment by kalessil** [comment]
Thank you. 

Also it worth updating to v3.0.0 (the issues will be still there, but the release has many bug-fixes).

**Comment by Bilge** [comment]
Thanks, but I don't know much about that. I thought PhpStorm was supposed to keep plugins updated automatically?

**Comment by kalessil** [comment]
No worries, PhpStorm has a popup about available updates (for IDE and plugins). Or these IDE settings: 
![image](https://user-images.githubusercontent.com/1577185/38214337-9dd2b194-36c4-11e8-9d24-85d1c431a2c3.png)

If the plugin has no update button, right-click on the list and click "Reload list of plugins"

**Comment by Bilge** [comment]
It does have a manual update button, but I don't want to manually update anything, I want automatic updates. For some reason that doesn't seem to be available.

**Comment by kalessil** [comment]
Ok =) I'll poke you when we have a solution for the issues itself.

**Comment by kalessil** [comment]
Fixed
